### ISSUE TITLE
TabPFN

### ISSUE THREAD

**Issue by szilard** [szilard/GBM-perf#62]
TabPFN: https://github.com/PriorLabs/TabPFN

Running TabPFN vs XGBoost and LightGBM on the airline data (on increasing training sample sizes).

Code here: [https://github.com/szilard/GBM-perf/tree/master/analysis/tabpfn](https://github.com/szilard/GBM-perf/tree/master/analysis/tabpfn)

### Run Time

TabPFN runs here on GPU (L4 on EC2) as it is very slow on CPU. XGBoost/LightGBM runs on CPU for simplicity.

TabPFN:

Train size | Time (s)
-------|--------
1000 |  1.8
2000 | 2.1
5000 | 3.2
10000 | 5.5
20000 | 11.8
50000 | 44.6
100000 | 146

XGBoost/LightGBM <1 sec for all sizes, 100x faster than TabPFN

### Accuracy

XGBoost/LightGBM not tuned, just some decent guess for the hyper-parameters 

AUC:

| Size   | TabPFN    | XGBoost   | LightGBM  |
|------|----------|----------|-----------|
| 1000   | **0.6964**| 0.6440    | 0.6475    |
| 2000   | **0.7072**| 0.6916    | 0.6874    |
| 5000   | 0.7082    | 0.7072    | **0.7074**|
| 10000  | **0.7096**| 0.7072    | 0.7075    |
| 20000  | 0.7165    | **0.7189**| 0.7182    |
| 50000  | 0.7228    | 0.7225    | **0.7239**|
| 100000 | 0.7250    | **0.7303**| **0.7303**|

XGBoost/LightGBM more accurate than TabPFN (even untuned) except for small data

Detailed results here: [https://github.com/szilard/GBM-perf/blob/master/analysis/tabpfn/results.txt](https://github.com/szilard/GBM-perf/blob/master/analysis/tabpfn/results.txt)

### Larger Data

TabPFN run time scales like O(N^2), not practical for larger data sizes.

XGBoost/LightGBM can do easily tens-hundreds of millions of rows
